pkg/ccl/streamingccl/streamproducer/stream_lifetime.go

```diff
@@ -307,3 +326,37 @@ func completeReplicationStream(
 		return nil
 	})
 }
+
+func setupSpanConfigsStream(
+	ctx context.Context, eval *eval.Context, txn isql.Txn, tenantName roachpb.TenantName,
+) (*streampb.ReplicationStreamSpec, error) {
+
+	tenantRecord, err := sql.GetTenantRecordByName(ctx, eval.Settings, txn, tenantName)
+	if err != nil {
+		return nil, err
+	}
+
+	tenantID := roachpb.MustMakeTenantID(tenantRecord.ID)
+
+	var spanConfigID descpb.ID
+	execConfig := eval.Planner.ExecutorConfig().(*sql.ExecutorConfig)
+	if err := sql.DescsTxn(ctx, execConfig, func(ctx context.Context, txn isql.Txn, col *descs.Collection) error {
+		g := col.ByName(txn.KV()).Get()
+		_, imm, err := descs.PrefixAndTable(ctx, g, systemschema.SpanConfigurationsTableName)
+		if err != nil {
+			return err
+		}
+		spanConfigID = imm.GetID()
+		return nil
+	}); err != nil {
+		return nil, err
+	}
+	codec := keys.MakeSQLCodec(roachpb.SystemTenantID)
+	spanConfigKey := codec.TablePrefix(uint32(spanConfigID))
+
+	// TODO(msbutler): crop this span to the keyspan within the span config
+	// table relevant to this specific tenant once I teach the client.Subscribe()
+	// to stream span configs, which will make testing easier.
+	span := roachpb.Span{Key: spanConfigKey, EndKey: spanConfigKey.PrefixEnd()}
+	return buildReplicationStreamSpec(ctx, eval, tenantID, true, roachpb.Spans{span})
```

Comment:
A couple of questions here:

1) In a recent conversation, I believe you and Arul pointed out to me that the span configuration table assumes some invariants.  Are we sure we are going to be able to deal with those invariants will be OK if we divide processing of the table up across. multiple nodes?  

2) Aside from the above, I wonder if we could get away without this?  Even if the span config table is split across multiple ranges, we could still stream it from a single source node, that source node would just have a couple remote rangefeeds potentially.  But, that doesn't seem _that_ bad to me.

Reply:
Good questions!! I needed to dig a bit more here.

This [comment](https://github.com/cockroachdb/cockroach/blob/00ec8d330a18d77e3aaa2e6d02e79f2d03a01404/pkg/spanconfig/spanconfig.go#L45) describes those invariants and this [little beast](https://github.com/cockroachdb/cockroach/blob/7a57e7ea1e4b188c76407edf4d318099a9f2bcbf/pkg/spanconfig/spanconfigstore/span_store.go#L284) of a function enforces them on all _proposed_ span config updates. For C2C to obey these invariants, I _believe_ we merely need to stream the point/range tombstones that the `kv.Accessor` writes to the source's span_configuration table, and not worry about manually splitting span config records based on what's already been committed.

To that end, C2C must stream all updates from the span_configuration table in the right order, even  if the table is split across ranges. (I should check with Arul if a given tenant's span config updates can get split across ranges, though I imagine this is allowed). More explicitly, if we write a tombstone on a tenant's span config update at t1, we must replicate that tombstone before replicating any updates later than t1, regardless of any simultaneous splits. 

A question for a rangefeed expert (you): Do we get this ordering guarantee for free, with the "rangefeed retry on splits" behavior, if one src node is running the show (i.e. your proposal in question 2)? Or would I need to add extra hwm timestamp management on the destination side?